### DISCUSSION TITLE
A question about plot subplots of network

### DISCUSSION THREAD

**Discussion by likeduck** [PMEAL/OpenPNM#2766] [Q&A] [answered]
Dear Prof, 
  I 'm sorry to bother you, although this is a matplotlib issue, it may involve the visualization module of openpnm. I would like to plot two images showing the relative sizes of pores and throats, but I encountered some issues. For example, I successfully removed the axes using the first method, but the two subplots overlapped. With the second method, the subplots did not overlap, but I couldn't remove the axes. So, I'm seeking your assistance and hoping for a solution.
  The first method is : 
```
fig, ax = plt.subplots(2, 1)
fig.subplots_adjust(hspace=0.2, wspace=0.1)
# Pore
op.visualization.plot_coordinates(network=net_coarse, pores=net_coarse.Ps, size_by=net_coarse['pore.diameter'], color_by=net_coarse['pore.diameter'], edgecolor='k', s=600, ax=ax[0])
op.visualization.plot_connections(network=net_coarse, throats=net_coarse.Ts, c='k', alpha=0.5, ax=ax[0])
ax[0].set_title('Pore Distribution', fontstyle='italic')
ax[0].axis('off')
# Throat
op.visualization.plot_coordinates(network=net_coarse, pores=net_coarse.Ps, c='k', s=50, ax=ax[1])
op.visualization.plot_connections(network=net_coarse, throats=net_coarse.Ts, linewidth=net_coarse['throat.diameter']*2, color_by=net_coarse['throat.diameter'], ax=ax[1])
ax[1].set_title('Throat Distribution', fontstyle='italic')
ax[1].axis('off')
plt.show()
```
![1](https://github.com/PMEAL/OpenPNM/assets/117707506/ee864a15-60da-4a51-8cb1-56f98e8e9459)

  The second method is : 
```
fig, ax = plt.subplots(2, 1)
fig.subplots_adjust(hspace=0.2, wspace=0.1)
# Pore
ax1 = fig.add_subplot(211)
op.visualization.plot_coordinates(network=net_coarse, pores=net_coarse.Ps, size_by=net_coarse['pore.diameter'], color_by=net_coarse['pore.diameter'], edgecolor='k', s=600, ax=ax1)
op.visualization.plot_connections(network=net_coarse, throats=net_coarse.Ts, c='k', alpha=0.5, ax=ax1)
ax1.set_title('Pore Distribution', fontstyle='italic')
# Throat
ax2 = fig.add_subplot(212)
op.visualization.plot_coordinates(network=net_coarse, pores=net_coarse.Ps, c='grey', s=50, ax=ax2)
op.visualization.plot_connections(network=net_coarse, throats=net_coarse.Ts, linewidth=net_coarse['throat.diameter']*2, color_by=net_coarse['throat.diameter'], ax=ax2)
ax2.set_title('Throat Distribution', fontstyle='italic')
plt.show()
```
![2](https://github.com/PMEAL/OpenPNM/assets/117707506/c669b106-6ae6-4f97-82ef-5dbfd4bfce4e)
  Although the second method code does not show the operation to remove the axes, please trust me that I have tried many approaches.
  I believe this might be due to a conflict between matplotlib and the visualization module of OpenPNM, but I'm not sure how to handle it. Could you provide me with some suggestions or ideas on how to address this issue?

**Comment by jgostick** [comment, marked as answer]
Hi @likeduck
It so happens that I have been playing with matplotlib a lot lately.  I have found this really helps:

```
h = op.visualization.plot_connection(...)
h = op.visualization.plot_coordinates(..., ax=h)

fig, ax = plt.gcf(), plt.gca()
< manipulate axes here >

```


This makes make things that our `ax` argument is not properly named since it's neigther and axis or a figure.  Anyway, hopefully this helps.  Once you have the `ax` and `fig` handles, you should be able to do whatever you like.

**Reply by likeduck** [reply]
Thanks! It really help me!

**Reply by jgostick** [reply]
great....I edited `plt, gca()` to `plt.gca()`.  I guess you figured that out.  Glad it helped.
